### PR TITLE
Implementes RoutingMessage::create_reply + few test fixes

### PR DESCRIPTION
**Pull request by inetic**
[Review on Reviewable](https://reviewable.io/reviews/maidsafe/routing/491)

### CODE REVIEW

**Comment by maidsafe-highfive** [comment]
r? @benjaminbollen

(maidsafe_highfive has picked a reviewer for you, use r? to override)

File: src/messages.rs [hunk 1]
```diff
@@ -224,21 +224,27 @@ impl RoutingMessage {
...
+        reply_message.destination = match self.orig_message {
+            Some(ref orig_message) => {
+                try!(orig_message.get_routing_message()).reply_destination()
+            },
+            None => {
+                self.reply_destination()
+            }
+        };
+
+        reply_message.orig_message = None;
```
Comment (benjaminbollen) [inline src/messages.rs]: Should the original message not be attached again when it is present?

Comment (inetic) [inline src/messages.rs]: It should, but not in `reply_message.orig_message`, instead it should be in `reply_message.message_type.<response-type>.orig_request`. So, currently it is the responsibility of the one who calls `create_reply` to set it explicitly.

Comment (benjaminbollen) [inline src/messages.rs]: you are right, I warned I found this highly confusing :)

**Comment by benjaminbollen** [comment]
looks good to me

**Pull request merged**